src/components/expansion-panel/expansion-panel.ts

```diff
@@ -0,0 +1,235 @@
+import { styles as bootstrap } from './themes/light/expansion-panel.bootstrap.css.js';
+import { styles as fluent } from './themes/light/expansion-panel.fluent.css.js';
+import { styles as indigo } from './themes/light/expansion-panel.indigo.css.js';
+
+let NEXT_ID = 0;
+export interface IgcExpansionPanelComponentEventMap {
+  igcOpening: CustomEvent<IgcExpansionPanelComponent>;
+  igcOpened: CustomEvent<IgcExpansionPanelComponent>;
+  igcClosing: CustomEvent<IgcExpansionPanelComponent>;
+  igcClosed: CustomEvent<IgcExpansionPanelComponent>;
+}
+
+const TABBABLE_SELECTORS =
+  'a[href], button, input, textarea, select, details, [tabindex]:not([tabindex="-1"])';
+
+/**
+ * The Expansion Panel Component provides a way to display information in a toggleable way -
+ * compact summary view containing title and description and expanded detail view containing
+ * additional content to the summary header.
+ *
+ * @element igc-expansion-panel
+ *
+ * @slot title - renders the title of the panel's header
+ * @slot subTitle - renders the subtitle of the panel's header
+ * @slot indicator - renders the expand/collapsed indicator
+ *
+ * @fires igcOpening - Emitted before opening the expansion panel.
+ * @fires igcOpened - Emitted after the expansion panel is opened.
+ * @fires igcClosing - Emitted before closing the expansion panel.
+ * @fires igcClosed - Emitted after the expansion panel is closed.
+ *
+ * @csspart header - The container of the expansion indicator, title and subTitle.
+ * @csspart title -  The title container.
+ * @csspart subTitle - The subTitle container.
+ * @csspart indicator - The indicator container.
+ * @csspart content - The expansion panel's content wrapper.
+ */
+@themes({ bootstrap, fluent, indigo })
+export default class IgcExpansionPanelComponent extends EventEmitterMixin<
+  IgcExpansionPanelComponentEventMap,
+  Constructor<LitElement>
+>(LitElement) {
+  public static readonly tagName = 'igc-expansion-panel';
+  public static styles = styles;
+
+  /** Indicates whether the contents of the control should be visible. */
+  @property({ reflect: true, type: Boolean })
+  public open = false;
+
+  /** Get/Set whether the expansion panel is disabled. Disabled panels are ignored for user interactions. */
+  @property({ reflect: true, type: Boolean })
+  public disabled = false;
+
+  /** The indicator position of the expansion panel. */
+  @property({ reflect: true, attribute: 'indicator-position' })
+  public indicatorPosition: 'start' | 'end' = 'start';
+
+  @query('[part~="header"]', true)
+  protected panelHeader!: HTMLElement;
+
+  private panelId!: string;
+
+  public override connectedCallback() {
+    super.connectedCallback();
+    const id = this.getAttribute('id');
+    this.panelId! = id ? id : 'igc-expansion-panel-' + ++NEXT_ID;
+  }
+
+  private handleClicked(event: Event) {
+    const el = event.target as HTMLElement;
+
+    if (!el.matches(TABBABLE_SELECTORS)) {
+      this.panelHeader!.focus();
+    }
```

Comment:
@damyanpetev raised a valid point over this code here.

What is the behavior that we try to achieve here? Shifting focus over to the header?
Isn't the header by default focusable if not disabled?

Reply:
We introduced this because when clicking on non-focusable elements, such as divs containing the title/subtitle slots and the text itself, the header does not seem to take focus. Consequently, as there is no focus after clicking in this case, keyboard navigation does not function properly.

Reply:
Seems to be something to do with the shadow DOM handling, otherwise in light DOM a _focusable_ element (that includes `div` with `tabindex`) will be [focused on click as well](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/tabindex#try_it).
I'm okay with the functionality staying the way it is, though the `TABBABLE_SELECTORS` approach is shaky - the list is far from exhaustive and we have our own components that are direct native replacements (eg `igc-button`). Doesn't seem like feasible/worth it chasing all possible elements. 
Considering the design of the component doesn't really cater to interactive content inside the header, I'm thinking we can probably do with an unconditional focus and leave interactive content to control handling by stopping propagation if needed.

Reply:
@damyanpetev 
Agreed. With the current approach we handle native controls but if for whatever reason I place a `igc-input` component
in the header we don't handle it. Replace with any other custom input. I'm inclined for an unconditional focus on click. That is users can handle the click event in the Light DOM  and stop propagation depending on the scenario rather than do partial handling.